apps/console/src/features/sms-providers/pages/sms-providers.tsx

```diff
@@ -0,0 +1,712 @@
+            vonageSecret: "",
+            vonageSender: ""
+        }
+
+    };
+    const [ smsProviderSettings, setSmsProviderSettings ] = useState<SMSProviderSettingsState>({
+        providerParams: defaultProviderParams,
+        selectedProvider: null
+    });
+    const isReadOnly: boolean = useMemo(() => !hasRequiredScopes(
+        featureConfig?.smsProviders,
+        featureConfig?.smsProviders?.scopes?.update,
+        allowedScopes
+    ), [ featureConfig, allowedScopes ]);
+
+    const {
+        data: originalSMSProviderConfig,
+        isLoading: isSMSProviderConfigFetchRequestLoading,
+        mutate: mutateSMSProviderConfig,
+        error: smsProviderConfigFetchRequestError
+    } = ListSMSProviders();
+
+    /**
+     * Displays the error banner when unable to fetch sms provider configuration.
+     */
+    const handleRetrieveError = (): void => {
+        dispatch(
+            addAlert({
+                description: t("extensions:develop.smsProviders." +
+                    "notifications.getConfiguration.error.description"),
+                level: AlertLevels.ERROR,
+                message: t("extensions:develop.smsProviders." +
+                    "notifications.getConfiguration.error.message")
+            })
+        );
+    };
+
+    /**
+     * Displays the sucess banner when sms provider configurations are updated.
+     */
+    const handleUpdateSuccess = () => {
+        dispatch(
+            addAlert({
+                description: t("extensions:develop.smsProviders." +
+                    "notifications.updateConfiguration.success.description"),
+                level: AlertLevels.SUCCESS,
+                message: t("extensions:develop.smsProviders." +
+                    "notifications.updateConfiguration.success.message")
+            })
+        );
+    };
+
+    /**
+     * Displays the error banner when unable to update sms provider configurations.
+     */
+    const handleUpdateError = () => {
+        dispatch(
+            addAlert({
+                description: t("extensions:develop.smsProviders." +
+                    "notifications.updateConfiguration.error.description"),
+                level: AlertLevels.ERROR,
+                message: t("extensions:develop.smsProviders." +
+                    "notifications.updateConfiguration.error.message")
+            })
+        );
+    };
+
+    /**
+     * Displays the sucess banner when sms provider configurations are deleted.
+     */
+    const handleDeleteSuccess = () => {
+        dispatch(
+            addAlert({
+                description: t("extensions:develop.smsProviders." +
+                    "notifications.deleteConfiguration.success.description"),
+                level: AlertLevels.SUCCESS,
+                message: t("extensions:develop.smsProviders." +
+                    "notifications.deleteConfiguration.success.message")
+            })
+        );
+    };
+
+    const [ isLoading, setIsLoading ] = useState(true);
```

Comment:
These should be re-ordered as follows.
1. React states and props
2. useEffect hooks
3. general functions (like `handleDeleteSuccess`)
4. Returned UI elements